backend/src/migrations/20240221213010_add_user_achievements_table.ts

```diff
@@ -0,0 +1,40 @@
+import { type Knex } from 'knex';
+
+const TABLE_NAME = 'user_achievements';
+
+const ColumnName = {
+    ID: 'id',
+    USER_ID: 'user_id',
+    ACHIEVEMENT_ID: 'achievement_id',
+    DATE_ACHIEVED: 'date_achieved',
+} as const;
+
+async function up(knex: Knex): Promise<void> {
+    return knex.schema.createTable(TABLE_NAME, (table) => {
+        table.increments(ColumnName.ID).primary();
+        table
+            .integer(ColumnName.USER_ID)
+            .unsigned()
+            .notNullable()
+            .references('id')
+            .inTable('users')
```

Comment:
let's replace string literals with enum values, for tables do the same as for columns
TableName = {...}

Reply:
let's also add index for this userId